analyzers/src/SonarAnalyzer.Common/Rules/StringLiteralShouldNotBeDuplicatedBase.cs

```diff
@@ -77,40 +77,25 @@ private void ReportOnViolation(SyntaxNodeAnalysisContext context)
             }
 
             var stringLiterals = RetrieveLiteralExpressions(context.Node);
-
-            // Collect duplications
-            var stringWithLiterals = new Dictionary<string, List<TLiteralExpressionSyntax>>();
-            foreach (var literal in stringLiterals)
-            {
-                // Remove leading and trailing double quotes
-                var stringValue = ExtractStringContent(GetLiteralValue(literal));
-
-                if (stringValue != null
-                    && stringValue.Length >= MinimumStringLength
-                    && !IsMatchingMethodParameterName(literal))
-                {
-                    if (!stringWithLiterals.ContainsKey(stringValue))
-                    {
-                        stringWithLiterals[stringValue] = new List<TLiteralExpressionSyntax>();
-                    }
-
-                    stringWithLiterals[stringValue].Add(literal);
-                }
-            }
+            var duplicateValuesAndPositions = from literal in stringLiterals
```

Comment:
Please rewrite it to lambda style, we don't use this syntax

Reply:
I did so in 2ecefbed9718f28ee5a59eee83c58d65565f2f50 I personally think that the comprehension query version is superior to the LINQ version. The range variables are so much easier to handle in nested scenarios and group by is so much easier to read. Should I put that on a "How we work"?

Reply:
You can. You should put two links there with the query version and the current version to show the idea.